src/inmanta/execute/runtime.py

```diff
@@ -785,6 +816,11 @@ def __init__(self, block: "BasicBlock", resolver: Resolver):
         self.block = block
         self.slots: Dict[str, ResultVariable] = {n: ResultVariable() for n in block.get_variables()}
         self.resolver = resolver
+        self.dataflow_graph: Optional[DataflowGraph] = None
+        if resolver.dataflow_graph is not None:
+            self.dataflow_graph = DataflowGraph(self, resolver.dataflow_graph)
+            for name, var in self.slots.items():
+                var.set_dataflow_node(self.dataflow_graph.get_named_node(name))
```

Comment:
This is where the binding of a node to a ResultVariable currently happens. Is this not what you meant @wouterdb?